### PR TITLE
ses: cmake: Added start and end tags in CMakeLists.txt

### PR DESCRIPTION
**Pull request by tejlmand**
To support adding / removal of:
- files
- include folders
- compile flags
of applications / samples in SES, the following tags must be present in
the CMakeLists.txt file of the app.
- `# NORDIC SDK APP START`
- `# NORDIC SDK APP END`

Signed-off-by: Torsten Rasmussen <torsten.rasmussen@nordicsemi.no>

### CODE REVIEW

File: samples/nrf9160/lte_ble_gateway/CMakeLists.txt [hunk 1]
```diff
@@ -20,7 +20,9 @@ set(spm_CONF_FILE
 include($ENV{ZEPHYR_BASE}/cmake/app/boilerplate.cmake NO_POLICY_SCOPE)
 project(lte-ble-gateway)
 
+# NORDIC SDK APP START
 target_sources(app PRIVATE src/main.c)
+# NORDIC SDK APP END
 zephyr_include_directories(src)
 
 include_directories(src/orientation_detector)
```
Comment (joerchan) [inline samples/nrf9160/lte_ble_gateway/CMakeLists.txt]: This include is not part of the SDK APP?

Comment (tejlmand) [reply samples/nrf9160/lte_ble_gateway/CMakeLists.txt]: No, cause it is not limited to the target `app`

SES only supports the following CMake functions in its integration to the IDE.
- `target_sources`
- `target_compile_definitions`
- `target_include_directories`
- `target_compile_options`

**Comment by tejlmand** [comment]
@ru-fu Added the description the fields `# NORDIC SDK APP START` and `# NORDIC SDK APP END` to the **Building with SES** section, as we don't have any other SES description place (at least I couldn't find it).

In future I assume we might want to describe / make a quick start guide on how to create your own sample app, or a guide for SES with NCS, as those could cover topics beyond getting started.
And when such docs are in place, the added section can be moved.

**Comment by ru-fu** [comment]
This is not where that content should go. I'll add a suggestion in a separate commit (will also fix the build error).

**Comment by tejlmand** [comment]
> will also fix the build error

The build error was due to latest `west` changes, and is fixed by locking west version

**Comment by ru-fu** [comment]
Sorry, I meant only the doc build error, won't fix anything else. ;)

**Comment by tejlmand** [comment]
> This is not where that content should go. 

That was my feeling as well, that this was not the right place.
But couldn't find a better place. Thus getting the content in place, then it's easier to relocate.

> I'll add a suggestion in a separate commit

Great, thanks :)

File: doc/nrf/gs_programming.rst [hunk 2]
```diff
@@ -86,6 +87,53 @@ You can download it from the following links:
    .. note::
    	In a multi-image build, this allows you to debug the source code of your application only.
 
+
+.. _gs_programming_manage:
+
+Managing projects with SES
+=========================
+
+Using SES it is possible to add/remove files to the application project using the SES.
```
Comment (mbolivar) [inline doc/nrf/gs_programming.rst]: The final "using the SES" is redundant and can be removed.

File: doc/nrf/gs_programming.rst [hunk 3]
```diff
@@ -86,6 +87,53 @@ You can download it from the following links:
...
+
+
+Adding files
+-----------------------
+
+To add files to a project, right click on the **Project 'app/libapp.a'** and select:
+- Add new file to CMakeLists.txt
+- Add existing file to CMakeLists.txt
+
+    .. figure:: images/ses_add_files.png
```
Comment (mbolivar) [inline doc/nrf/gs_programming.rst]: Did you mean for this figure to be under the bullet point, or at top level? If top level, the leading whitespace should be removed.

In general I think the usual style in .rst is to put top level directives at column 0, so this applies elsewhere in the patch.

File: doc/nrf/gs_programming.rst (hunk 2 above)
Comment (tejlmand) [reply doc/nrf/gs_programming.rst]: Thanks, will check up, when @ru-fu makes her doc updates

File: doc/nrf/gs_programming.rst (hunk 3 above)
Comment (tejlmand) [reply doc/nrf/gs_programming.rst]: correct, will remove indentation.

File: doc/nrf/gs_programming.rst [hunk 4]
```diff
@@ -86,6 +87,53 @@ You can download it from the following links:
...
+
+    - ``# NORDIC SDK APP START``
+    - ``# NORDIC SDK APP END``
+
+must be surrounding the following CMake commands in the application CMakeLists.txt file.
+
+    - ``target_sources``
+    - ``target_compile_definitions``
+    - ``target_include_directories``
+    - ``target_compile_options``
```
Comment (SebastianBoe) [inline doc/nrf/gs_programming.rst]: I assume this is just for the 'app' library?

If so, this should be stated.

One can define multiple libraries in the application CMakeLists.txt file.

File: doc/nrf/gs_programming.rst [hunk 5]
```diff
@@ -86,6 +87,53 @@ You can download it from the following links:
...
+
+
```
Comment (SebastianBoe) [inline doc/nrf/gs_programming.rst]: What about invocations of zephyr_library_sources() in the application CMakeLists.txt file?

File: doc/nrf/gs_programming.rst [hunk 6]
```diff
@@ -86,6 +87,53 @@ You can download it from the following links:
...
+Adding files
+-----------------------
```
Comment (SebastianBoe) [inline doc/nrf/gs_programming.rst]: What happens if there is an invocation

target_source(app PRIVATE main.c)

outside of the tags?

Will the user simply not be able to remove the file through the IDE, or are there more serious consequences?
(Corrupt build)

Perhaps this level of detail is not something a user cares about, but I would like to know ...

**Comment by ru-fu** [comment]
I updated the documentation - please check and squash if okay.
Ping me if you need the link to look at the output.

File: doc/nrf/gs_programming.rst (hunk 5 above)
Comment (tejlmand) [reply doc/nrf/gs_programming.rst]: No, that is currently not supported by SES.

Only the CMake functions stated are supported.

File: doc/nrf/gs_programming.rst (hunk 4 above)
Comment (tejlmand) [reply doc/nrf/gs_programming.rst]: Good point.
To be made clearer.

File: doc/nrf/gs_programming.rst (hunk 6 above)
Comment (tejlmand) [reply doc/nrf/gs_programming.rst]: invocation outside tags is what we have today without this PR.
So the answer is simply that they will be ignored by the IDE.

**Comment by SebastianBoe** [comment]
This change is documentation and comments, so it is safe to merge during feature-freeze.

**Pull request merged**